Remove kategorie from seed entry and convert variety to a simple text field.

badnames opened the pull request:
This pull requests implements the following changes requested by @markus2330 in #35:


> completely remove "Kategorie" (seeds and plants database). So the very first entry field should be Art, then Sorte, then Year.

> Variety, i.e., Sorte later will be select-able from database but for the first milestone it is simply text to be entered (As I assume we will not be able to implement the "Edit Plants" use case next week, which would be needed to add the varieties.)

markus2330 commented on backend/migrations/2023-03-03-140707_convert_variety_to_text/up.sql:
```diff
@@ -0,0 +1,10 @@
+-- Add column without not null constraint.
+ALTER TABLE seeds ADD COLUMN variety TEXT;
+-- Then set all varieties to some default value .... 
+UPDATE seeds SET variety = 'Default';
+-- ... before setting not null constraint.
+ALTER TABLE seeds ALTER COLUMN variety SET NOT NULL;
```
Variety of seeds can be NULL. This represents that the the variety of the plant_id is used.

markus2330 reviewed (changes requested):
Great job! Thank you for the cleanup!

- please update the README.md from backend (there is no varieties anymore)
- http://127.0.0.1:5173/seeds nor curl 192.168.178.24:8080/api/seeds shows the seeds I just entered? Is it really inserted in the DB? This worked before.

badnames commented:
Thank you for the review @markus2330.

> please update the README.md from backend (there is no varieties anymore)
Yes I missed that. Its now fixed. 

> http://127.0.0.1:5173/seeds nor curl 192.168.178.24:8080/api/seeds shows the seeds I just entered? Is it really inserted in the DB? This worked before.

I could not replicate this on my machine. Requests from the frontend, as well as plain http requests sent using curl resulted in new database entries.
Perhaps there was a bug somewhere that was now fixed by accident?

Please verify that this issue is still present with the latest changes and send the relevant curl command if the issue persists.

> Documentation/frontend-tests could be improved

I agree, but I would prefer to open another pull request for this issue as it is unrelated to #35 in my opinion.

markus2330 commented:
> I could not replicate this on my machine.

My suspicion is that my backend was wrongly configured. Afaik the frontend does not give an error if the connection fails, so I did not notice. I added this to #35.

We now merged the PR with the new plant table (replacing varieties). This created quite a few conflicts here. Let us discuss today in which order we will finish the first milestone.

badnames commented:
I was just resolving the merge conflicts and encountered a problem.

As it is, both the selction field for plants and for variety is called "Sorte". 
How should these textfields be renamend?

badnames commented:
I have noticed that the new seed property plant_id from the master branch is also called "Sorte" now. 
How should we rename  the variety field to distinguish these two properties?

badnames commented:
Ok, thank you for the answer.
I merged the changes from master into this branch by using two text fields for now.

After the build issues are fixed, I will look into merging both text fields as requested.

badnames commented:
> There is a double use for Sorte (German)/variety (English) field:

>    if entered in the Seed entry manually, the Seed database gets an entry variety and this is preferred
>    the "Plant" is a variety, then the data is used from the Plant database (variety in seed database is NULL in this case)

Users now get a single selectbox/typeahead that allows them to create items that are not in the list. If a new item is created, it will be stored in the variety field of the seed table, otherwise the seed_id will be set as before.

markus2330 commented:
@lukashartl do you know when we will be able to deploy PRs? It would be handy here if I don't need to test locally.

markus2330 commented:
Great job! Looking forward to other fixes of #35.
